.github/workflows/api-dotnetcore.yml

```diff
@@ -7,7 +7,24 @@ on:
     branches: [master, dev, test]
 
 jobs:
+  # JOB to run change detection
+  check-changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
```

Comment:
I don't see that this is only affecting pull requests, am I missing something?

Reply:
This was a comment on the sample for that filter. I believe it will work properly on pushes too. On that matter, do you know when that would be the case? AFAIK changes to dev, test or master can only be done though PRs

Reply:
Yeah, the comment confused me because it is possible to restrict what workflows trigger an action. So you could restrict this to run on PRs but not pulls for example. I don't see that happening here however.